test/model/UserSpec.scala

```diff
@@ -0,0 +1,52 @@
+package model
+
+import org.scalatest.matchers.should.Matchers.shouldBe
+import org.scalatestplus.play.PlaySpec
+import play.api.libs.json.Json
+
+class UserSpec extends PlaySpec {
+
+  "writes.me" should {
+    "generate correct Json response for /user/me" in {
+      val user = User(
+        identityId = "123",
+        brazeId = None,
+        emailAddress = Some("test@user"),
+        emailValidated = Some(true),
+        userName = Some("test@user"),
+        name = Name(title = Some("Mx"), firstName = Some("Test"), secondName = Some("User")),
+        address = Address(
+          line1 = Some("1"),
+          line2 = Some("2"),
+          line3 = Some("3"),
+          line4 = Some("4"),
+          postcode = Some("5"),
+          country = Some("United Kingdom"),
+        ),
+        phoneNumber = Some(PhoneNumber(44, "12345")),
+        registrationLocation = Some("online"),
+        permissions = Seq(Permission("p1", true), Permission("p2", false), Permission("p3", true)),
+      )
+      Json.stringify(Json.toJson(user)(User.writes.me)) shouldBe
+        """{
+          |"identityId":"123",
+          |"emailAddress":"test@user",
+          |"emailValidated":true,
+          |"username":"test@user",
+          |"title":"Mx",
+          |"firstName":"Test",
+          |"secondName":"User",
+          |"addressLine1":"1",
+          |"addressLine2":"2",
+          |"addressLine3":"3",
+          |"addressLine4":"4",
+          |"postcode":"5",
+          |"country":"United Kingdom",
+          |"countryCode":44,
+          |"localNumber":"12345",
+          |"registrationLocation":"online",
+          |"permissions":[{"id":"p1","permitted":true},{"id":"p2","permitted":false},{"id":"p3","permitted":true}]
+          |}""".stripMargin.replace("\n", "")
+    }
+  }
```

Comment:
It might be useful to generate use cases with empty fields to make sure we understand how that would work.  For example would fields set to empty strings be represented in a different way than empty fields ?

Reply:
Actually, there would never be an empty field at this point because they get [stripped out](https://github.com/guardian/gatehouse/blob/7ef263c3ef99da22de0084ee32b4dd9a0d18a606/app/services/LegacyIdentityDbUserService.scala#L54-L71) when they go into the `User` model.
[Example](https://github.com/guardian/gatehouse/blob/7ef263c3ef99da22de0084ee32b4dd9a0d18a606/test/services/LegacyIdentityDbUserServiceSpec.scala#L21).

Reply:
so what would the json look like if the user has fields set to `None`?  would the fields be missing from the json ? Maybe we could have a test case to show that and make sure it doesn't change in the future ?

Reply:
Test case in 568235d697d5d105cc2caca43ed8ee2ed8ac8f15